certora/specs/HashConsensus.spec

```diff
@@ -0,0 +1,814 @@
+//      setFrameConfig() - MANAGE_FRAME_CONFIG_ROLE 
+//      setFastLaneLengthSlots() - MANAGE_FAST_LANE_CONFIG_ROLE
+//      addMember() - MANAGE_MEMBERS_AND_QUORUM_ROLE
+//      removeMember() - MANAGE_MEMBERS_AND_QUORUM_ROLE
+//      setQuorum() - MANAGE_MEMBERS_AND_QUORUM_ROLE, DISABLE_CONSENSUS_ROLE (only if disabling it)
+//      disableConsensus() - MANAGE_MEMBERS_AND_QUORUM_ROLE, DISABLE_CONSENSUS_ROLE
+//      setReportProcessor() - MANAGE_REPORT_PROCESSOR_ROLE
+//      submitReport() - only if getIsMember(msg.sender) == true
+//  3. setFrameConfig() updates epochsPerFrame in a way that either keeps the current reference slot
+//     the same or increases it by at least the minimum of old and new frame sizes
+//  4. setFastLaneLengthSlots() works as expected, setting it to zero disables the fast lane subset
+//     verify with getFrameConfig() and check that fastLaneLengthSlots < frameConfig.epochsPerFrame * SLOTS_PER_EPOCH
+//  5. addMember() - cannot add an existing member
+//  6. addMember() - cannot add an empty address as a member
+//  7, addMember() - adding a member does not remove any other member
+//  8. addMember() - adding a member increases the total members by 1
+//  9. removeMember() - cannot remove a member that does not exists - reverts
+// 10. removeMember() - removing a member does not remove any other member
+// 11. removeMember() - removing a member decreases the total members by 1
+// 12. setQuorum() - acts as expectedly, verified by getQuorum()
+// 13. disableConsensus() - acts as expectedly, verified by getQuorum()
+// 14. setReportProcessor() - cannot set an empty address
+// 15. submitReport() - slot > max_uint64 => revert
+// 16. submitReport() - slot < currentRefSlot => revert
+// 17. submitReport() - slot <= lastProcessingRefSlot => revert
+// 18. submitReport() - reportHash == 0 => revert
+// 19. submitReport() - consensusVersion != getConsensusVersion() => revert
+// 20. submitReport() - verify that _computeTimestampAtSlot(frame.reportProcessingDeadlineSlot) > TimeOf(frame.refSlot)
+//                             that the deadline is explicitly == TimeOf(refslot+FrameSize)
+// 21. submitReport() - revert if same oracle reports the same slot+hash (cannot double vote for same report)
+// 22. submitReport() - single call to submit report increases only one support for one variant
+//                      sum of supports of all variants < total members,
+//                      because if Oracle member changes its mind, its previous support is removed
+// 23. submitReport() - all variants should be removed when new frame starts
+
+
+//  1. All the external view functions cannot revert under any circumstance,
+//     especially getCurrentFrame() as it is used by LegacyOracle and BaseOracle
+// Status: Fail
+// https://vaas-stg.certora.com/output/80942/985192ce8baf4725a8b6d29c1d0dc2af/?anonymousKey=f2574e5803ece9f5b8d734ecd68b1d9ac714c1d5
+rule viewFunctionsDoNotRevert(method f)
```

Comment:
I suppose that if you intend verifying this rule, you want to include the sane assumptions block you've created?